packages/react/src/modules/slack/components/SlackChannelCombobox/SlackChannelListBox.tsx

```diff
@@ -0,0 +1,48 @@
+import { SlackChannel, SlackChannelConnection } from "@knocklabs/client";
+import { AriaListBoxOptions } from "@react-aria/listbox";
+
+import SlackChannelOption from "./SlackChannelOption";
+import "./styles.css";
+
+type Props = AriaListBoxOptions<unknown> & {
+  slackChannels: SlackChannel[];
+  isLoading: boolean;
+  connectedChannels: SlackChannelConnection[] | null;
+  onClick: (channelId: string) => void;
+  listBoxProps?: React.HTMLAttributes<HTMLDivElement>;
+  channelOptionProps?: React.HtmlHTMLAttributes<HTMLButtonElement>;
+};
+
+const SlackChannelListBox = ({
+  slackChannels,
+  isLoading,
+  connectedChannels,
+  onClick,
+  listBoxProps,
+  channelOptionProps
+}: Props) => {
+  return (
+    <div className="rnf-list-box" {...listBoxProps}>
+      {slackChannels.map((channel) => {
+        return (
+          <SlackChannelOption
+            tabIndex={0}
+            channel={channel}
+            isLoading={isLoading}
+            isConnected={
+              !!connectedChannels?.find(
+                (connectedChannel) =>
+                  connectedChannel.channel_id === channel.id,
+              )
+            }
+            onClick={onClick}
+            channelOptionProps={channelOptionProps}
+
+          />
+        );
+      })}
```

Comment:
Should add a `key` to `SlackChannelOption` to help react render this long list.
```suggestion
      {slackChannels.map((channel) => {
        return (
          <SlackChannelOption
            key={channel.id}
            tabIndex={0}
            channel={channel}
            isLoading={isLoading}
            isConnected={
              !!connectedChannels?.find(
                (connectedChannel) =>
                  connectedChannel.channel_id === channel.id,
              )
            }
            onClick={onClick}
            channelOptionProps={channelOptionProps}

          />
        );
      })}
```